PR: https://github.com/odoo/odoo/pull/234974

From: 113d77eb35aa8f8b503043d1201b6cdf78e9be83
From: Hesham Saleh
Date: 2025-12-24 09:57:26

Breaking data model changes scores: del:8 + add:0, change matches:
-    bank_street = fields.Char(related='bank_id.street', readonly=False)
-    bank_street2 = fields.Char(related='bank_id.street2', readonly=False)
-    bank_zip = fields.Char(related='bank_id.zip', readonly=False)
-    bank_city = fields.Char(related='bank_id.city', readonly=False)
-    bank_state = fields.Many2one(related='bank_id.state', readonly=False)
-    bank_country = fields.Many2one(related='bank_id.country', readonly=False)
-    bank_email = fields.Char(related='bank_id.email', readonly=False)
-    bank_phone = fields.Char(related='bank_id.phone', readonly=False)

Total Changes: 12

[IMP] *: Improve UX of bank accounts

This commit improves the UX of bank accounts by:
1. removing res.bank model
2. removes localizations redundancy
3. some fields in localizations should be available to all companies
4. most countries had missing fields

task-5145618

closes odoo/odoo#234974

Related: odoo/enterprise#99418
Related: odoo/upgrade#8958
Related: odoo/upgrade-util#358
Signed-off-by: Habib Ayob (ayh) <ayh@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/res_partner_bank.py
+++ b/addons/hr/models/res_partner_bank.py
@@ -6,18 +6,10 @@ from odoo import api, fields, models
 class ResPartnerBank(models.Model):
     _inherit = 'res.partner.bank'
 
-    bank_street = fields.Char(related='bank_id.street', readonly=False)
-    bank_street2 = fields.Char(related='bank_id.street2', readonly=False)
-    bank_zip = fields.Char(related='bank_id.zip', readonly=False)
-    bank_city = fields.Char(related='bank_id.city', readonly=False)
-    bank_state = fields.Many2one(related='bank_id.state', readonly=False)
-    bank_country = fields.Many2one(related='bank_id.country', readonly=False)
-    bank_email = fields.Char(related='bank_id.email', readonly=False)
-    bank_phone = fields.Char(related='bank_id.phone', readonly=False)
     employee_id = fields.Many2many('hr.employee', 'Employee', compute="_compute_employee_id", search="_search_employee_id")
     employee_salary_amount = fields.Float(string='Salary Allocation', compute='_compute_salary_amount', digits=(16, 4), readonly=True, store=False)
     employee_salary_amount_is_percentage = fields.Boolean(compute='_compute_salary_amount', readonly=True, store=False)
-    currency_symbol = fields.Char(related='currency_id.symbol')
+    currency_symbol = fields.Char(related='employee_id.currency_id.symbol')
     employee_has_multiple_bank_accounts = fields.Boolean(related="employee_id.has_multiple_bank_accounts")
 
     @api.depends('employee_id.salary_distribution')
@@ -54,5 +46,5 @@ class ResPartnerBank(models.Model):
             account_employee = self.sudo().filtered("partner_id.employee_ids")
             for account in account_employee:
                 account.sudo(self.env.su).display_name = \
-                    account.acc_number[:2] + "*" * len(account.acc_number[2:-4]) + account.acc_number[-4:]
+                    account.account_number[:2] + "*" * len(account.account_number[2:-4]) + account.account_number[-4:]
         super(ResPartnerBank, self - account_employee)._compute_display_name()
